### ISSUE TITLE
[SUGGESTION] Remove ship hp

### ISSUE THREAD

**Issue by Leha44581** [tsunamayo/Starship-EVO#2913]
Please remove ship hp system so it won't be destroyed if all systems (reactors, thrusters or weapons) is still operational
Example why ship hp system is bad:
![20200906141720_1](https://user-images.githubusercontent.com/70238465/92324625-2e1c2a80-f04c-11ea-9ff9-b9e74127d895.jpg)
![20200906141725_1](https://user-images.githubusercontent.com/70238465/92324626-2eb4c100-f04c-11ea-934f-9369c48a9f13.jpg)
![20200906141119_1](https://user-images.githubusercontent.com/70238465/92324627-2f4d5780-f04c-11ea-8c48-68fe5583cc95.jpg)
![20200906141128_1](https://user-images.githubusercontent.com/70238465/92324628-2f4d5780-f04c-11ea-8020-4d504ad7120a.jpg)
Here i get just one hole and ran out of hp but don't lose anything except armor, and it is kinda stupid that you can destroy any ship by just scratching it armor

**Comment by Whackjob** [comment]
That's my Arkingthaad!  Go, baby, go!

I agree 100,000%.   My ship has massive amounts of redundancies so once done it could've lost dozens of reactors and still kept fighting.  Except, not with this system.  Reduction of volume only shouldn't be enough.

**Comment by tsunamayo** [comment]
This system is in place so you dont have to fire for 30min at a ship that is looking like a cheese. That being said it seems not well calibrated for large ship, could you please send your blueprint to me (export the destroyed one!)? I wonder if that could be a bug with some system chip. Thanks

**Comment by ProPeach** [comment]
This is a great example of why the hp system needs more balance, not why it needs to be removed in my opinion. The recently introduced system hp system is a step in the right direction, but it would work even better if it were combined with the hull hp. So a ship would have an overall hp (structural integrity, whatever) that is drained by destroying hull, but drained much much more by destroying system blocks. I think it's also a lot simpler in terms of understanding and GUI too

**Comment by Leha44581** [comment]
[Blueprints.zip](https://github.com/tsunamayo/Starship-EVO/files/5179582/Blueprints.zip)
@tsunamayo 
sure
Also here is another screenshot
![20200906142650_1](https://user-images.githubusercontent.com/70238465/92325086-86552b80-f050-11ea-9ca4-7c369607019b.jpg)

**Comment by AlienXtream** [comment]
i dont think we should worry about this at the moment. as of right now the infrastructure side of ships is basically nonexistent and that may play a big role in the final form of ship HP. imo this is a crude but passable substitute until ship infrastructure (like fuel, items, life support, etc) are figured out and implemented. once there is "more stuff to break" then ship damage and HP could be changed to work with that but until then i just dont think its worth the effort to work out a new system only to have it totally replaced later when the current one works well enough till then. thats just my view ¯\_(ツ)_/¯

**Comment by Servomotor** [comment]
The issue has less to do with ship HP, and more to do with ridiculous usage of weaponry and weapon balance overall. The current meta is to build a gun and attach a ship to it, and anything but beam weapons is pretty much useless right now. 

IMO each weapon attached to a child entity should increase the energy consumption exponentially. Game balance exists so that both players in an pvp engagement have an opportunity for survival, either by killing the opponent or by fleeing. It might be fun being the guy/gal with the one shot gun who only builds for the pvp meta, but its not fun for the guy/gal who builds for role play/ship character and to make a fun usage of the building system the entire game is designed around in the first place, who has no chance against "one shot" builds. 

If we are going to allow 16x16 fully stretched beam weapons to a single turret, floating in mid air so there is unrestricted of enemy engagement, then expect a game build around nothing but 16x16 fully stretched beam weapon brick ships. 

Games can not simulate every single variable that accounts for why a design is made the way it is like in real life, this is why restrictions have to be implemented, to help account for those variables we can not simulate.

**Comment by Whackjob** [comment]
I'm just speaking for myself here, and clearly I'm biased, sure, but it doesn't seem wise to have an environment uniquely suited to make capital ship combat at least possible, and then etterly whittle it down to blandness just to appease people who aren't interested in that sort of thing and wouldn't indulge in it anyway.  Let big capital ships with lots of guns remain possible, introduce armor at some point I think, but then in the future just make a server setting limit ship size, should that be desired by the operator.  That way the rest of us aren't hamstrung by people not even taking part in it.  They get what they want, we get what we want, and then I'm not interfering with them, and they're not constricting me.

**Comment by Claakin** [comment]
Perhapse a sensor could be created, make it a bit difficult to use such as the L for light cruise with the charge up, but longer. Assign a key to it such as 1, 2, 3, etc, or program in a sensor button like the 'Primary' or 'Landing Gear' buttons, but make it 'Sensors.' 

That way if you can target a ship, and are able to scan it in time, maybe the systems become visible for a time being? Maybe highlighted by the systems UI white box for 10 seconds. Ship systems visibility could allow for a more focused attack and relieve the need to remove health, and keep away from swiss cheese attacks. Making this interactive with a need for the sensors module and the press and hold of a button to scan could be a creative way to go about it.

**Comment by WILDCATreactor** [comment]
What about... let's call them "integrity points" your hull is worth X/Y, each reactor worth A/B, and each console, weapon, thruster, shield generator, or other secondary systems is worth Z/C points or whatever math ends up working.  Hull and reactor systems contribute the most Integrity Points, the other sub-systems are just there to add flesh to the !HP system.  Once a ship loses [whatever] percentage of "IP" its agility is downgraded, so light craft can lose lots of IP, proportionally to their size, and still function, if they only lose surface features and redundant systems- they can still limp along until critical damage to hull or reactors pushes them past the "Capital Grade" agility bracket.  This also means hull blocks versus bricks are an important design choice in light and ultralight craft, bricks provide no IP, but Hull blocks do- having no hull blocks means any reactor hits will result in critical performance losses.

Once a Ship's IP loss pushes its Agility to 0% or lower it loses power to engines, so a big ship can sustain lots of damage, lose lots of IP, and still be able to fight back despite being hobbled until repairs can be effected.  A ship inside of a gravity well will fall when hobbled.  Similarly, decent hits to fighter-sized vessels could hypothetically one-shot their IP without entirely destroying them.  Armored core systems and spaced redundancies become important then, not just for surviving general damage but to contribute to the IP pool.  I mean, yes, you could make an extremely distributed ship that is annoying to kill, so maybe larger single-block systems are more efficient for volume:weight ratios compared to smaller "block arrays" and redundancies, so you generally want a strong core system bolstered by redundancies and not a design made up entirely of them. 
 In the same way larger hull blocks would be better than many of them, emphasizing the need for a "ship spine".  Consequently, building a ship mostly from Hull blocks would grant it a large Integrity Point pool, but any damage would be likely to degrade it as well, if only fractionally.

- Hull and Reactor blocks provide a large number of Integrity Points
- Weapons, shields, and other "subsystems" provide a small number of IP
- IP scales geometrically with Hull and Reactor block size, so a few large blocks beats many smaller blocks of the same volume
- Subsystem IP scales linearly with volume
- As IP is reduced by destroying blocks that have an IP value, ship Agility degrades
- Once a ship receives repairs that increase its IP, Agility is restored per the same formula (centralised systems are easier to provide damage control to)
- At XX% IP a ship is 'Hobbled', Hobbled ships have Zero Agility and engine systems provide little to no thrust, Lightcruise does not function
- Ships inside of a gravity well will fall while Hobbled.

More or less, it's HP that causes a death spiral due to Agility loss and can't be directly attacked by weapons (rather it's depleted by system or hull loss), and you can still fight even while at zero !HP, you're just a very easy target and crew should consider getting to the lifeboats or making critical repairs instead of making a glorious last stand.  Attackers might dislike having to deal with a hobbled battleship, but as a sitting duck, ripping out its remaining weapons and boarding to clear out the survivors is easier than a partly mobile ship full of holes and uncertain subsystem locations.  Further, capturing a hobbled ship and repairing it is no easy feat, but a few key repairs can get it moving again after the medbay/respawn and armory has been dealt with.  Salvaging battlezones for derelicts becomes a bit more viable as well, since while repairs are easy now, having a resource system active means players will need to make strategic choices about what to repair in the field and what to save for when they get home to the shipyards.  These strategic choices also carry over to shipboard engineering and DCon teams, as a ship will only be carrying so many resources for repairs, it's not necessarily wise to patch every hole in the ship all the time, and material should be conserved for critical repairs until the battle is over.
